Lib/glyphsLib/builder.py

```diff
@@ -340,6 +344,10 @@ def set_custom_params(ufo, parsed=None, data=None, misc_keys=(), non_info=()):
         if name.startswith('openTypeOS2Win') and value < 0:
             value = -value
 
+        # Some Glyphs files store weightClass/widthClass as string values.
+        if name in ('openTypeOS2WeightClass', 'openTypeOS2WidthClass'):
+            value = int(value)
```

Comment:
Oh, now I see. The lines I linked above refer to the types of the `weightClass` and `widthClass` *attributes* of the `instances` element in the glyphs file structure.
Those should stay `string` since, as you pointed out, their value is an enum.

What I meant was to cast the `weightClass` and `widthClass` *custom parameters* to integer as we read them in, by adding them to the `CUSTOM_INT_PARAMS` set at the top of `casting.py`.

Note that in the handbook the `weightClass` and `widthClass` custom parameters are said to be of *integer* type:
https://glyphsapp.com/content/1-get-started/2-manuals/1-handbook-glyphs-2-0/Glyphs-Handbook-2.3.pdf#page=202